PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:9 + add:0, change matches:
-    _inherit = "product.product"
-    _inherit = 'product.template'
-    _inherit = 'stock.lot'
-    _inherit = 'procurement.group'
-    _inherit = 'res.config.settings'
-    _inherit = "stock.move"
-    _inherit = "stock.move.line"
-    _inherit = "stock.picking"
-    _inherit = 'stock.quant'

Total Changes: 20

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/product_expiry/models/product_product.py
+++ b/addons/product_expiry/models/product_product.py
@@ -4,8 +4,8 @@
 from odoo import fields, models
 
 
-class Product(models.Model):
-    _inherit = "product.product"
+class ProductProduct(models.Model):
+    _inherit = ["product.product"]
 
     def action_open_quants(self):
         # Override to hide the `removal_date` column if not needed.
@@ -15,7 +15,7 @@ class Product(models.Model):
 
 
 class ProductTemplate(models.Model):
-    _inherit = 'product.template'
+    _inherit = ['product.template']
 
     use_expiration_date = fields.Boolean(string='Use Expiration Date',
         help='When this box is ticked, you have the possibility to specify dates to manage'

--- a/addons/product_expiry/models/production_lot.py
+++ b/addons/product_expiry/models/production_lot.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, SUPERUSER_ID, _
 
 
 class StockLot(models.Model):
-    _inherit = 'stock.lot'
+    _inherit = ['stock.lot']
 
     use_expiration_date = fields.Boolean(
         string='Use Expiration Date', related='product_id.use_expiration_date')
@@ -90,7 +90,7 @@ class StockLot(models.Model):
 
 
 class ProcurementGroup(models.Model):
-    _inherit = 'procurement.group'
+    _inherit = ['procurement.group']
 
     @api.model
     def _run_scheduler_tasks(self, use_new_cursor=False, company_id=False):

--- a/addons/product_expiry/models/res_config_settings.py
+++ b/addons/product_expiry/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     group_expiry_date_on_delivery_slip = fields.Boolean("Display Expiration Dates on Delivery Slips",
         implied_group='product_expiry.group_expiry_date_on_delivery_slip')

--- a/addons/product_expiry/models/stock_move.py
+++ b/addons/product_expiry/models/stock_move.py
@@ -10,7 +10,7 @@ from odoo.tools import get_lang
 
 
 class StockMove(models.Model):
-    _inherit = "stock.move"
+    _inherit = ["stock.move"]
 
     use_expiration_date = fields.Boolean(
         string='Use Expiration Date', related='product_id.use_expiration_date')

--- a/addons/product_expiry/models/stock_move_line.py
+++ b/addons/product_expiry/models/stock_move_line.py
@@ -8,7 +8,7 @@ from odoo.tools.sql import column_exists, create_column
 
 
 class StockMoveLine(models.Model):
-    _inherit = "stock.move.line"
+    _inherit = ["stock.move.line"]
 
     expiration_date = fields.Datetime(
         string='Expiration Date', compute='_compute_expiration_date', store=True,

--- a/addons/product_expiry/models/stock_picking.py
+++ b/addons/product_expiry/models/stock_picking.py
@@ -5,7 +5,7 @@ from odoo import models, _
 
 
 class StockPicking(models.Model):
-    _inherit = "stock.picking"
+    _inherit = ["stock.picking"]
 
     def _pre_action_done_hook(self):
         res = super()._pre_action_done_hook()

--- a/addons/product_expiry/models/stock_quant.py
+++ b/addons/product_expiry/models/stock_quant.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class StockQuant(models.Model):
-    _inherit = 'stock.quant'
+    _inherit = ['stock.quant']
 
     expiration_date = fields.Datetime(related='lot_id.expiration_date', store=True)
     removal_date = fields.Datetime(related='lot_id.removal_date', store=True)
